### PR TITLE
Fix Vault write and update commands

### LINKED ISSUE #2000: vault update -command gets stuck at least with certain parameters
Action:
* "kontena vault update --upsert test_var 1"
Expected result:
* test_var variable is created or updated with value 1 into vault
Observed result:
* Command prompt gets stuck and the action is not performed

Other:
* kontena version
  * cli: 1.1.5
  * master: 1.1.4
* Bug confirmed by @nevalla

### PR DESCRIPTION
**Pull request by nevalla**
This PR fixes `kontena vault update` command and unifies logic with `kontena vault write` command. Also added unit tests for both commands.

Fixes #2000

### CODE REVIEW

File: cli/spec/kontena/cli/vault/update_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+require_relative "../../../spec_helper"
+require 'kontena/cli/vault/update_command'
+
+describe Kontena::Cli::Vault::UpdateCommand do
+
+  include RequirementsHelper
+  include ClientHelpers
+
+  let(:subject) do
```
Comment (SpComb) [inline cli/spec/kontena/cli/vault/update_spec.rb]: The `ClientHelpers` should already do all of this boilerplate, down to the `allow(subject).to receive(:current_grid).and_return('test-grid')`?

File: cli/spec/kontena/cli/vault/update_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  describe '#execute' do
+    before(:each) do
+      allow(subject).to receive(:client).with(token).and_return(client)
+      allow(subject).to receive(:current_grid).and_return('test-grid')
+    end
+
+    it 'returns error if value not provided' do
+      allow(STDIN).to receive(:read).once.and_return('')
+      expect(subject).to receive(:exit_with_error).with('No value provided').and_call_original
```
Comment (SpComb) [inline cli/spec/kontena/cli/vault/update_spec.rb]:
```ruby
      expect{subject.run(['mysql_password'])}.to exit_with_error.and output(/No value provided/).to_stderr
```

File: cli/spec/kontena/cli/vault/update_spec.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
+require_relative "../../../spec_helper"
```
Comment (SpComb) [inline cli/spec/kontena/cli/vault/update_spec.rb]: Not needed anymore.

**Comment by SpComb** [comment]
Looks like this overlaps with #1936 @kke

**Comment by kke** [comment]
Replaced with #2045

**Pull request closed without merging**